__test__/integrationTests.test.ts

```diff
@@ -120,6 +156,35 @@ describe("Integration tests for AUTH routes:", () => {
     // Reject invalid inputs
     // Reject unexpected attributes
     // Others
+    // user validation test
+    describe("user validation test : ", () => {
+        test.each([
+            { missingFieldName: "userName" },
+            { missingFieldName: "email" },
+            { missingFieldName: "password" },
+        ])(
+            "return 400 when $missingFieldName field is missing",
+            async ({ missingFieldName }) => {
+                const user: RegistrationData = createRandomUserForRegister();
+
+                delete user[missingFieldName as keyof typeof user];
+
+                const response = await axios.post("api/auth/register", user);
+
+                expect(response.status).toBe(400);
+            }
+        );
+
+        test("return 400 when password is too short", async () => {
+            const user: RegistrationData = {
+                ...createRandomUserForRegister(),
+                password: "123",
+            };
+
+            const response = await axios.post("api/auth/register", user);
+            expect(response.status).toBe(400);
+        });
+    });
```

Comment:
This is a good start. We will need to add more tests once the password pattern is in place though to make sure it is working as intended.

Reply:
Thanks, Jacob.